include/aspect/particle/world.h

```diff
@@ -83,6 +83,7 @@ namespace aspect
          * take the properties of the closest particle, but more complicated
          * schemes are possible.
          *
+         * @param [in] generator The particle generator for this world.
          * @param [in] manager The property manager for this world.
          * @param [in] load_balancing The strategy used to balance the
          * computational load for particle advection across processes.
```

Comment:
in the function declaration below this comment you pass a lot of `shared_ptr` objects by value. that's almost certainly not wrong, but it doesn't do anything useful unless (i) you call the function with a raw pointer, and (ii) you actually store the resulting shared_ptr object somewhere. In other words, you *capture* a pointer. If that's what you do, it would be easier to read if you just passed the raw pointer. 

But I suspect that that's not what you do, and in that case you'll probably be better off taking these objects as const references.